lib/spack/spack/environment.py

```diff
@@ -1486,6 +1510,11 @@ def install_all(self, args=None, **install_args):
 
         try:
             builder = PackageInstaller(installs)
+
+            # If the user has requested a monitor client, attach to builder
+            if install_args.get('use_monitor', False):
```

Comment:
I wish we had a single place to put this and other preinstall stuff.  But maybe this is fine for now.

Reply:
Yeah it's probably not ideal to be passing around install-args! What did you have in mind?